Meziantou.Polyfill.Editor/M;System.Linq.Enumerable.ToDictionary``2(System.Collections.Generic.IEnumerable{System.Collections.Generic.KeyValuePair{``0,``1}},System.Collections.Generic.IEqualityComparer{``0}).cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+
+static partial class PolyfillExtensions
+{
+    public static Dictionary<TKey, TValue> ToDictionary<TKey, TValue>(
+        this IEnumerable<KeyValuePair<TKey, TValue>> source,
+        IEqualityComparer<TKey>? comparer)
+            where TKey : notnull
+    {
+        if (source is null)
```

Comment:
I think you should optimize it for .NET Core 2+

````c#
#if NETCOREAPP2_0_OR_GREATER
return new Dictionary<TKey, TValue>(source, comparer);
#else
// Your current implementation
#endif
````

https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.-ctor?view=net-9.0&WT.mc_id=DT-MVP-5003978#system-collections-generic-dictionary-2-ctor(system-collections-generic-ienumerable((system-collections-generic-keyvaluepair((-0-1))))-system-collections-generic-iequalitycomparer((-0)))